heron/simulator/src/java/com/twitter/heron/simulator/instance/SpoutInstance.java

```diff
@@ -78,7 +79,7 @@ protected void produceTuple() {
       }
 
       if (collector.getTotalDataEmittedInBytes() - totalDataEmittedInBytesBeforeCycle
```

Comment:
The collector and the simulator both still track how many bytes they've emitted as a long, so I convert to that and do the comparison. I've mainly just been converting the static `ByteAmounts` that are read from the configs, but not the dynamic ones that get frequently updated at runtime. I'd like to also update the latter but I need to look at that more closely to make sure we can always do that efficiently without too much object creation, versus incrementing a long. I'd like to hold on that for a follow-up PR.